crates/web-client/test/new_transactions.test.ts

```
@@ -0,0 +1,417 @@
+        let secret_key = account_auth.get_rpo_falcon_512_secret_key_as_felts();
+        let transcription_script_input_pair_array = new window.TransactionScriptInputPairArray([new window.TransactionScriptInputPair(public_key, secret_key)]);
+        let transaction_script = await client.compile_tx_script(tx_script, transcription_script_input_pair_array);
+        let note_id = note.id();
+        let note_args_commitment = window.Rpo256.hash_elements(feltArray); // gets consumed by NoteIdAndArgs
+        let note_id_and_args = new window.NoteIdAndArgs(note_id, note_args_commitment.to_word());
+        let note_id_and_args_array = new window.NoteIdAndArgsArray([note_id_and_args]);
+        let advice_map = new window.AdviceMap();
+        let note_args_commitment_2 = window.Rpo256.hash_elements(feltArray);
+        advice_map.insert(note_args_commitment_2, feltArray);
+
+        let transaction_request_2 = new window.TransactionRequest()
+            .with_authenticated_input_notes(note_id_and_args_array)
+            .with_custom_script(transaction_script)
+            .extend_advice_map(advice_map)
+
+        // Execute and Submit Transaction
+        await client.fetch_and_cache_account_auth_by_pub_key(walletAccount.id());
+        let transaction_result_2 = await client.new_transaction(walletAccount.id(), transaction_request_2);
+        await client.submit_transaction(transaction_result_2);
+        await new Promise(r => setTimeout(r, 10000));
+        await client.sync_state();
+    },
+    asserted_value);
+};
+
+describe("custom transaction tests", () => {
+    it("custom transaction completes successfully", async () => {
+        const result = await customTransaction("0");
+
+        expect(1).to.equal(1);
+    });
+
+    // TODO: Need better error handling throughout the new_transaction 
+    // and submit_transaction web-client call stacks to actually detect 
+    // this. Otherwise it hangs the test.
```

I had this problem of hanging tests while testing the new tags in one of my PRs, do we know why they happen?

---

This test in particular was written by @dagarcia7 before he went on leave , so I don't have the full context / haven't repro'd

That being said my guess is the puppeteer context might be still active after a `.wasm` error since its effectively just hosting a web page. I'll add an item for us to investigate propagating these errors up from the puppeteer context to the mocha tests so that things fail fast.